packages/fxa-auth-server/lib/routes/validators.js

```diff
@@ -293,11 +294,10 @@ module.exports.subscriptionsSubscriptionValidator = isA.object({
     .date()
     .timestamp('unix')
     .required(),
-  ended_at: isA.alternatives(
-    isA.date().timestamp('unix'),
-    isA.any().allow(null)
-  ),
-  nickname: isA.string().required(),
+  end_at: isA.alternatives(isA.date().timestamp('unix'), isA.any().allow(null)),
+  failure_code: isA.string().optional(),
+  failure_message: isA.string().optional(),
+  plan_name: isA.string().required(),
```

Comment:
Are we able to restrict the characters allowed in these strings, even if only to `routes.validators.DISPLAY_SAFE_UNICODE` or similar? I've seen too many bugs allowed through by the generic `.string()` validator...

Reply:
Someday we can, yes.